### ISSUE TITLE
Integrating the training of the base model

### ISSUE THREAD

**Issue by 2006pmach** [Megvii-BaseDetection/GFSD#4]
Hi,

I noticed that you use the pretrained base-class model form another repo. In the end, I guess it would also be possible to train it using `CVPODS` by adding another `exp` folder and the required files to the `playground` folder. Is there a simple way to achieve this? Would be nice to have everything at one place. And would be good example to start developing new Few-Shot Methods using your framework and dataset splits etc. What would I have to add? Or could you add this? Thx in advance.

**Comment by zb1439** [comment]
Thanks for the advice. This can be done by adjusting the training schedule, resnet depth, and input image scales to be identical to fsdet's repo. You can modify config.py in https://github.com/Megvii-BaseDetection/cvpods/tree/master/playground/detection/coco/rcnn/faster_rcnn.res50.fpn.coco.multiscale.1x to make sure it is the same as the config in fsdet (https://github.com/ucbdrive/few-shot-object-detection/blob/master/configs/COCO-detection/faster_rcnn_R_101_FPN_base.yaml and https://github.com/ucbdrive/few-shot-object-detection/blob/master/configs/Base-RCNN-FPN.yaml). I might try adding this experiment folder if I am sure the configuration is identical and reproducible.

**Comment by amajee11us** [comment]
@zb1439 can the base model trained using the FsDet framework be directly used during the finetuning process using GFSD?

**Comment by zb1439** [comment]
> @zb1439 can the base model trained using the FsDet framework be directly used during the finetuning process using GFSD?

Yes. They should be using the same name for weights. And this open-source implementation is intended to use their pretrained model as stated in readme.

**Comment by amajee11us** [comment]
Thanks for the update @zb1439 !!
FsDet performs a "surgery" operation on the base trained checkpoint through this script - https://github.com/ucbdrive/few-shot-object-detection/blob/master/tools/ckpt_surgery.py .
For GFSD, should the input model be the one after this surgery operation or the one before the surgery operation?

**Comment by zb1439** [comment]
> Thanks for the update @zb1439 !! FsDet performs a "surgery" operation on the base trained checkpoint through this script - https://github.com/ucbdrive/few-shot-object-detection/blob/master/tools/ckpt_surgery.py . For GFSD, should the input model be the one after this surgery operation or the one before the surgery operation?

The input pretrained model does not need to be trimmed. Once you have that pretrained weight, you are all set to fine-tune it using this repo.
